test_reproducibility.py

```diff
@@ -29,19 +37,21 @@ def tcase_csv_names():
     ]
 
 
-def compare_results(gt_dir: Path, test_dir: Path, csv_names: list[str]) -> None:
+def compare_results(gt_dir: Path, test_dir: Path, csv_names: list[str], experiment_step: str) -> None:
     for csv_name in csv_names:
-        gt_df = pd.read_csv(gt_dir / csv_name, index_col=0)
+        gt_df = pd.read_csv(gt_dir / experiment_step / csv_name, index_col=0)
         test_df = pd.read_csv(test_dir / csv_name, index_col=0)
         pd.testing.assert_frame_equal(gt_df, test_df, obj=csv_name)
         print(f"Test passed for {csv_name}")
 
 
-def test_e2e(tcase_config, tcase_csv_names, tmpdir):
-    tcase_config["logging"]["out_dir"] = tmpdir
-    set_seed(tcase_config["run"]["random_seed"])
-    main_runner.run_experiments(tcase_config)
-    compare_results(Path("_test_data"), Path(tmpdir), tcase_csv_names)
+def test_e2e(tcase_experiment_steps, tcase_config, tcase_csv_names, tmpdir):
+    for experiment_step in tcase_experiment_steps:
+        tcase_config["run"]["experiment_step"] = experiment_step
+        tcase_config["logging"]["out_dir"] = str(tmpdir)
+        set_seed(tcase_config["run"]["random_seed"])
+        main_runner.run_experiments(tcase_config)
+        compare_results(Path("_test_data"), Path(tmpdir), tcase_csv_names, experiment_step)
```

Comment:
(3) Tests don't pass for the tensor runner. Did you write the code using a workstation with CUDA? I think that can be a reason for this problem. If so, I'd recommend to disable CUDA, e.g. by setting this env. variable: `export CUDA_VISIBLE_DEVICES=""`

Reply:
![Screenshot 2024-07-14 at 17 27 35](https://github.com/user-attachments/assets/576b2ce6-41d5-4630-bfef-8347a50c7e03)